src/tmpfiles/tmpfiles.c

```diff
@@ -2261,12 +2310,64 @@ static int remove_item(Item *i) {
         }
 }
 
+static int age_by_format(AgeBy age_by, char **ret_age_by_fmt) {
+        _cleanup_free_ char *tmp = NULL;
+        _cleanup_strv_free_ char **split = NULL;
+        char types[5] = { 0 };
+        char *p = NULL, *q = NULL;
+
+        assert(ret_age_by_fmt);
+
+        if (age_by == AGE_BY_UNSET) {
+                *ret_age_by_fmt = strdup("");
+                if (!*ret_age_by_fmt)
+                        return -ENOMEM;
+
+                return 0;
+        }
+
+        p = types;
+        if (age_by & AGE_BY_ATIME)
+                *p++ = 'a';
+
+        if (age_by & AGE_BY_BTIME)
+                *p++ = 'b';
+
+        if (age_by & AGE_BY_CTIME)
+                *p++ = 'c';
+
+        if (age_by & AGE_BY_MTIME)
+                *p++ = 'm';
+
+        /* Stop and rewind back to the beginning. */
+        *p = 0;
+        p = types;
+
+        do {
+                if (asprintf(&q, "%ctime", *p) < 0)
+                        return -ENOMEM;
+
+                if (strv_push(&split, q) < 0)
+                        return -ENOMEM;
+        } while(*(++p));
+
+        tmp = strv_join(split, ", ");
+        if (!tmp)
+                return -ENOMEM;
+
+        if (asprintf(ret_age_by_fmt, "(age-by: %s)", tmp) < 0)
+                return -ENOMEM;
+
+        return 0;
```

Comment:
i don't grok this function. first you generate the short version, and then you convert that to the long version? why? why not just generate the long version if you want it?

Reply:
I wanted to avoid creating new strings and `strv_push` in each `if` condition and instead went with this route.

Reply:
Changed this part quite a bit in the latest commit.